enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,299 @@
+
+#### oc <new|start>-build
+
+`oc new-build` should print a warning message if the generated/referenced BuildConfig uses the Jenkins pipeline strategy.
+This message should indicate that the cluster must have the DeprecatedJenkinsPipeline feature set enabled.
+Similarly, `oc start-build` should print a warning message if a build was started with the Jenkins pipeline strategy indicating that the cluster must have the DeprecatedJenkinsPipeline feature set enabled.
+
+#### Web Console
+
+User will continue to be able to deploy Jenkins on OpenShift by instantiating our provided Templates.
+As long as the Jenkins Pipeline strategy can be run on a fully supported cluster (i.e upgrades are allowed), existing console integrations with Jenkins should remain in place.
+
+#### Telemetry / Operational Readiness
+
+The feature set will take advantage of existing telemetry tooling.
+Clusters that enable the DeprecatedJenkinsPipeline feature set will be automatically reported to Telemeter.
+Fleet operators/SREs will be able to identify which clusters have the feature set enabled.
+
+The bootstrap behavior will ensure that most clusters which rely on the JenkinsPipeline build strategy have the feature set enabled automatically.
+The exception to this rule will be clusters that have a separate feature set enabled prior to upgrade.
+The following query in Data Hub will reveal which clusters fall into this category:
+
+`count by (_id, name) (cluster_feature_set{name!=""}) * on (_id) group_right() openshift:build_by_strategy:sum{strategy="jenkinspipeline"}`
+
+As of this writing, no clusters fall into this category.
+This may change if clusters opt into tech preview features related to OpenShift builds.
+
+#### Documentation
+
+Documentation related to the Jenkins Pipeline build strategy will need to inform admins that the `DeprecatedJenkinsPipeline` feature set needs to be enabled for builds to succeed.
+
+Examples for Jenkins usage in openshift/origin should be removed.
+Tests which utilized this example deployment should instead instantiate one of the Jenkins templates installed via the Samples operator.
+
+### Risks and Mitigations
+
+**Risk**: Jenkins pipeline builds fail on cluster upgrade, requiring manual intervention.
+
+*Mitigation*: openshift-controller-manager-operator will include bootstrap logic to detect if the cluster uses Jenkins Pipeline builds.
+
+**Risk**: Bootstrap logic can overwrite other enabled features.
+
+*Mitigation*: Bootstrap logic is only invoked if the cluster uses the default feature set.
+
+**Risk:** Clusters need tech preview features AND the Jenkins pipeline build strategy enabled.
+
+*Mitigation*: Clusters with tech preview features are automatically marked as `NoUpgrade` clusters.
+If necessary, a cluster that needs tech preview features and the Jenkins pipeline build strategy can use the `CustomNoUpgrade` feature set,
+which gives admins the ability to enable individual feature gates on their own.
```

Comment:
CustomNoUpgrade is really really hard to get right (I had a long discussion with @deads2k about this).

Can't they just enable both the TechPreviewNoUpgrade featureset *and* the DeprecatedJenkins featureset?